alerts.tf

```diff
@@ -23,65 +23,48 @@ resource "newrelic_synthetics_monitor" "health_check" {
 
 # urgent conditions
 
-resource "newrelic_synthetics_alert_condition" "health_check" {
+resource "newrelic_nrql_alert_condition" "health_check" {
   count = var.service_healthcheck_url != null ? 1 : 0
 
-  policy_id = newrelic_alert_policy.urgent.id
-
-  name        = "${var.newrelic_app_name}: health check"
-  monitor_id  = newrelic_synthetics_monitor.health_check[0].id
-  runbook_url = var.runbook_url
-}
-
-resource "newrelic_nrql_alert_condition" "error_rate" {
-  count = var.alert_error_rate_enable ? 1 : 0
-
   account_id = var.newrelic_account_id
   policy_id  = newrelic_alert_policy.urgent.id
 
-  name        = "${var.newrelic_app_name}: too many sustained errors"
+  name        = "${var.newrelic_app_name}: health check"
   runbook_url = var.runbook_url
   enabled     = true
 
   aggregation_method = "event_flow"
+  aggregation_window = 60
   aggregation_delay  = 180
   slide_by           = 30
 
   violation_time_limit_seconds = 86400
 
   critical {
-    operator              = "above"
-    threshold             = var.alert_error_rate_threshold
-    threshold_duration    = var.alert_error_rate_duration
+    operator              = "below"
+    threshold             = newrelic_synthetics_monitor.health_check[0].period_in_minutes * 60
+    threshold_duration    = var.alert_health_check_duration
     threshold_occurrences = "ALL"
   }
 
-  # The operation 'percentage(count(*), WHERE error IS TRUE)' can be represented as follows:
-  #
-  # Transaction events with error
-  # –––––––––––––––––––––––––––––
-  # Transaction events (overall)
-  #
-  # For intervals with no events the operation ends up being 0 / 0, which returns NULL.
-  # To avoid this problem we use a query that ensures the denominator is always a non-zero value.
   nrql {
     query = <<-EOF
-        SELECT 100 * (
-          filter(count(*), WHERE error IS TRUE) / (count(*) + 1e-10)
-        ) as Percentage
-        FROM Transaction
-        WHERE appName = '${var.newrelic_fully_qualified_app_name}'
+        SELECT count(*)
```

Comment:
Have you tested this works? This query returns a complex object. So, I just wonder whether the alert condition is doing the right thing with its result.

Reply:
No, I've not tested it. I saw it was possible but in NewRelic's forum, but I cannot find it now 🤦 .
I'll try to test this. Do you have any idea on how I can do that (other than providing a fake health url)?

Reply:
I tested it and I had to fix the query 😀 

As for the `FACET` part, NewRelic applies the threshold to each time series.

Reply:
I found it by the way: https://forum.newrelic.com/s/hubtopic/aAX8W0000008bEUWAY/synthetics-concurrent-monitor-failure-alerting-using-nrql-alerting

Reply:
I wouldn't use anything from a forum (unless it's very clear what it does), specially if it's not coming from a NR's employee.

The only documentation I have found is this (https://docs.newrelic.com/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/create-nrql-alert-conditions/#example-null) and this (https://docs.newrelic.com/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/create-nrql-alert-conditions/#example-null-zero).

Why do we need the `FACET` here?

One option you can consider is to manually (in the UI) create an alert for a synthetic monitor, and then see if it's possible to see the NRQL query.

Reply:
I already did that and tested the query with the `FACET`. The `FACET` groups results by location and the threshold applies to all locations as stated by the UI:
![image](https://github.com/vistaprint/terraform-newrelic-monitoring/assets/36991286/8fa9edc4-5289-49b2-9253-2d47bee9d05a)

But if you rather prefer not counting by locations (so behaving as if there was only one) I can make the change.